src/java.desktop/share/classes/javax/swing/plaf/basic/BasicFileChooserUI.java

```diff
@@ -709,16 +709,17 @@ public void valueChanged(ListSelectionEvent evt) {
                         if (objects.length == 1
                             && ((File)objects[0]).isDirectory()
                             && chooser.isTraversable(((File)objects[0]))
-                            && (useSetDirectory || (!fsv.isFileSystem(((File)objects[0]))
-                            && !Files.isSymbolicLink(Paths.get(((File)objects[0]).getPath()))))) {
+                            && (useSetDirectory
+                                || (!fsv.isFileSystem(((File)objects[0]))
+                                    && !Files.isSymbolicLink(((File)objects[0]).toPath())))) {
                             setDirectorySelected(true);
                             setDirectory(((File)objects[0]));
                         } else {
                             ArrayList<File> fList = new ArrayList<File>(objects.length);
                             for (Object object : objects) {
                                 File f = (File) object;
```

Comment:
I'd be grateful if you update this line even though it's not part of your changes:

```suggestion
                                File f = (File)object;
```

It's to make casts consistent. Only this cast has a space before the argument in the entire method.